chromium_src/chrome/browser/ui/translate/translate_bubble_model_impl.cc

```diff
@@ -0,0 +1,182 @@
+// Copyright 2013 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "chrome/browser/ui/translate/translate_bubble_model_impl.h"
+
+#include <utility>
+
+#include "chrome/browser/translate/chrome_translate_client.h"
+#include "components/translate/core/browser/language_state.h"
+#include "components/translate/core/browser/translate_ui_delegate.h"
+
+TranslateBubbleModelImpl::TranslateBubbleModelImpl(
```

Comment:
copying an entire class like this is not ok, this leads to upstream changes that get lost. What are you actually changing here?

Reply:
This comment is related to an obsolete version of file that wasn't finished. Now only the necessary methods are modified.